src/daft-core/src/array/from.rs

```diff
@@ -340,3 +340,30 @@ where
         )
     }
 }
+
+impl<T: NumericNative> From<(&str, Vec<Option<Vec<T>>>)> for ListArray
+where
+    T::DAFTTYPE: DaftNumericType<Native = T>,
+    ArrayWrapper<DataArray<T::DAFTTYPE>>: SeriesLike,
+{
+    fn from(item: (&str, Vec<Option<Vec<T>>>)) -> Self {
+        let (name, data) = item;
+        let flat_child_vec = data.iter().flatten().flatten().copied().collect::<Vec<_>>();
+        let flat_child = DataArray::<T::DAFTTYPE>::from((name, flat_child_vec)).into_series();
+
+        let lengths = data.iter().map(|d| d.as_ref().map_or(0, |d| d.len()));
+        let offsets = arrow2::offset::Offsets::try_from_lengths(lengths)
+            .unwrap()
+            .into();
+
+        let validity =
+            arrow2::bitmap::Bitmap::from_trusted_len_iter(data.iter().map(Option::is_some));
+
+        Self::new(
+            flat_child.field().to_list_field().rename(name),
+            flat_child,
+            offsets,
+            Some(validity),
+        )
+    }
+}
```

Comment:
This is maybe personal preference but I don't like the existing pattern of implementing `From` on a tuple like this, since it is pretty unclear that the first value is the name. Also having all of these implementations of `From` makes it difficult to figure out which one is being called for a certain tuple. I've been trying to move away from this pattern.

Reply:
yeah I agree. The `::from`'s make it kinda hard to find the implementation. I just did it this way because that's what was there already. Will create a dedicated method.